### PR TITLE
Added helix model and examples

### PR DESCRIPTION
**Pull request by jfrederik-nrel**
- Added helix wake mixing model
- Added helix turbine power loss model
- Added examples to showcase new functionality

To do's:
- [x] Format code @misi9170 
- [x] Write tests
- [x] Format examples
- [x] Add default tuning parameters for different turbines

### CODE REVIEW

File: examples/inputs/emgauss.yaml [hunk 1]
```diff
@@ -47,7 +47,8 @@ wake:
     velocity_model: empirical_gauss
 
   enable_secondary_steering: false
-  enable_yaw_added_recovery: true
+  enable_yaw_added_recovery: false
```
Comment (misi9170) [inline examples/inputs/emgauss.yaml:50]: Should `enable_yaw_added_recovery` be changed here to `false`? Ideally, we'd be able to leave this as `true`

File: examples/inputs/emgauss_iea_15mw.yaml
Comment (misi9170) [inline examples/inputs/emgauss_iea_15mw.yaml:1]: I'd prefer not to have to add an extra input file---would it instead be possible to add the `helix_wake_exp` and `helix_wake_denominator` terms to examples/inputs/emgauss.yaml, load from that input instead in example 42, and then switch the `turbine_type` to the IEA 15MW?

File: examples/42_test_helix.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+
+"""
+Example to test out using helix wake mixing of upstream turbines.
+Helix wake mixing is turned on at turbine 1, off at turbines 2 to 4;
+Turbine 2 is in wake turbine 1, turbine 4 in wake of turbine 3.
+"""
+
+# Grab model of FLORIS and update to helix-enabled turbines
+fmodel = FlorisModel("inputs/emgauss_iea_15mw.yaml")
```
Comment (misi9170) [inline examples/42_test_helix.py:30]: Could we instead add the necessary Helix parameters to inputs/emgauss.yaml; load inputs/emgauss.yaml; and then `set` the `turbine_types` to `["iea_15mw"]`? That would save us adding another example input yaml to keep track of.

**Review by misi9170 — CHANGES_REQUESTED** [review]
@jfrederik-nrel I've noted a handful of small updates that I think would clean this up. 

Then, a slightly bigger question that we have discussed before would be to change the names of `helix_amplitudes` and `helix_frequencies` to be more generic so that they can be used in future wake mixing models (or, if even more generic, perhaps other ways of operating the turbine that are not necessarily for the purpose of wake mixing).

A couple of options would be:
1. `wake_mixing_amplitude` and `wake_mixing_frequency`
2. `blade_pitch_amplitude` and `blade_pitch_frequency` 
3. `pitch_induced_mixing_amplitude` and `pitch_induced_mixing_frequency`

Others might have suggestions for this too!

Changing the names will be a little annoying as they appear in many places throughout Floris, but hopefully find and replace will be pretty effective once we decide on the names...

**Comment by misi9170** [comment]
Another note is that we'll also need to figure out how to handle the helix setpoints in the `UncertainFlorisModel` and `ParallelFlorisModel`. I'd say that for now, it's appropriate for those to simply raise `NotImplementedError`s if helix setpoints are passed in (which I can implement), but it would be good to get others' opinions on that.

File: examples/42_test_helix.py (hunk 2 above)
Comment (jfrederik-nrel) [reply examples/42_test_helix.py:30]: I get where you're coming from, but for now I've left it in for the following reason: the helix parameters are tuned specifically for 15MW simulations, so it feels weird to me to define these helix expansion parameters in a file defined for a turbine for which they aren't calibrated.

Comment (misi9170) [reply examples/42_test_helix.py:30]: That makes sense. Perhaps, then, we could instead call this file emgauss_helix.yaml, and have it specify the IEA_15MW as it's turbine (as it already does)? We could then put in the description (or just a comment at the top of the file) that the helix parameters specified are tuned to work with the IEA 15MW turbine.

File: floris/turbine_library/nrel_5MW.yaml [hunk 3]
```diff
@@ -39,6 +39,12 @@ power_thrust_table:
   cosine_loss_exponent_tilt: 1.88
   # Cosine exponent for power loss due to yaw misalignment.
   cosine_loss_exponent_yaw: 1.88
+  # Helix parameters
+  helix_a: 1.802
+  helix_power_b: 4.568e-03
+  helix_power_c: 1.629e-10
```
Comment (rafmudaf) [inline floris/turbine_library/nrel_5MW.yaml:45]: This is a very small number - is the helix model sensitive enough to changes at that order of magnitude? In other words, could this be 0?

File: tests/conftest.py [hunk 4]
```diff
@@ -214,6 +214,11 @@ def __init__(self):
                 "cosine_loss_exponent_tilt": 1.88,
                 "ref_air_density": 1.225,
                 "ref_tilt": 5.0,
+                "helix_a": 1.809,
```
Comment (rafmudaf) [inline tests/conftest.py:217]: These are different from the values in the NREL 5MW file. Should they be consistent?

File: floris/turbine_library/nrel_5MW.yaml (hunk 3 above)
Comment (jfrederik-nrel) [reply floris/turbine_library/nrel_5MW.yaml:45]: I agree that it's small, but it is multiplied by `base_powers`, which is in the order of 1e6. As a result, this sum is more or less the same magnitude as `helix_power_b`.

File: floris/core/solver.py [hunk 5]
```diff
@@ -1290,6 +1335,17 @@ def empirical_gauss_solver(
                 downstream_distance_D[:,:,i],
                 model_manager.deflection_model.yaw_added_mixing_gain
             )
+        # if model_manager.enable_active_wake_mixing:
```
Comment (paulf81) [inline floris/core/solver.py:1338]: Delete commented code?

**Review by misi9170 — APPROVED** [review]

**Comment by rafmudaf** [comment]
This pull request was squashed into https://github.com/NREL/floris/commit/9da4cbb39d06a7c7d2fb6af1c8f2c5b5192bacc2.

**Pull request merged**